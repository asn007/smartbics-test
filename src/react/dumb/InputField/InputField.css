:local .input {
  border: 1px solid #1e1e1e;
  border-radius: 3px;
  padding: 5px;
}

:local .input:focus {
  border: 1px solid black;
}
