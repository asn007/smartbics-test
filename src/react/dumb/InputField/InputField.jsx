import React from 'react';
import PropTypes from 'prop-types';
import style from './InputField.css';

export default function InputField({ className, ...rest }) {
  return <input className={style.input + ' ' + className} {...rest} />;
}

InputField.propTypes = {
  className: PropTypes.string
};

InputField.defaultProps = {
  className: ''
};
