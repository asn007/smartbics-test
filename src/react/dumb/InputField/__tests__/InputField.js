import React from 'react';
import InputField from '../InputField';
import theme from '../InputField.css';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('InputField', () => {
  test('it should render its markup correctly', () => {
    const component = <InputField className="test" value="text" />;
    const shallowRendered = shallow(component);
    const reactRendered = renderer.create(component);
    expect(reactRendered.toJSON()).toMatchSnapshot();
    expect(shallowRendered.hasClass(theme.input)).toBe(true);
    expect(shallowRendered.hasClass('test')).toBe(true);
  });
});
