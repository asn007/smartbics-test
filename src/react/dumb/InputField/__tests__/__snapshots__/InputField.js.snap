// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField it should render its markup correctly 1`] = `
<input
  className="input test"
  value="text"
/>
`;
